pkg/controller/nmstate/nmstate_controller.go

```diff
@@ -145,11 +147,28 @@ func (r *ReconcileNMState) applyRBAC(instance *nmstatev1beta1.NMState) error {
 
 func (r *ReconcileNMState) applyHandler(instance *nmstatev1beta1.NMState) error {
 	data := render.MakeRenderData()
+
+	// Prepare defaults
+	masterExistsNoScheduleToleration := corev1.Toleration{
+		Key:      "node-role.kubernetes.io/master",
+		Operator: corev1.TolerationOpExists,
+		Value:    string(corev1.TaintEffectNoSchedule),
+	}
+	webhookNodeSelector := map[string]string{
+		"beta.kubernetes.io/arch":        "amd64",
+		"node-role.kubernetes.io/master": "",
+	}
+
 	data.Data["HandlerNamespace"] = os.Getenv("HANDLER_NAMESPACE")
 	data.Data["HandlerImage"] = os.Getenv("HANDLER_IMAGE")
 	data.Data["HandlerPullPolicy"] = os.Getenv("HANDLER_IMAGE_PULL_POLICY")
 	data.Data["HandlerPrefix"] = os.Getenv("HANDLER_PREFIX")
-	data.Data["HandlerNodeSelector"] = instance.Spec.NodeSelector
+	data.Data["WebhookNodeSelector"] = helper.ToYaml(webhookNodeSelector)
+	data.Data["WebhookTolerations"] = helper.ToYaml([]corev1.Toleration{masterExistsNoScheduleToleration})
+	data.Data["WebhookAffinity"] = helper.ToYaml(corev1.Affinity{})
+	data.Data["HandlerNodeSelector"] = helper.ToYaml(instance.Spec.NodeSelector)
+	data.Data["HandlerTolerations"] = helper.ToYaml([]corev1.Toleration{masterExistsNoScheduleToleration})
+	data.Data["HandlerAffinity"] = helper.ToYaml(corev1.Affinity{})
```

Comment:
Create a little functoin here that do the `ToYaml` + the golang counterpart of  nindent 8 and use that functions at thise values.

Reply:
Thanks for your guidance here